src/dbnode/persist/fs/retriever.go

```diff
@@ -468,24 +501,22 @@ func (r *blockRetriever) fetchBatch(
 	}
 }
 
-func (r *blockRetriever) Stream(
+func (r *blockRetriever) streamRequest(
 	ctx context.Context,
+	req *retrieveRequest,
 	shard uint32,
 	id ident.ID,
 	startTime time.Time,
-	onRetrieve block.OnRetrieveBlock,
+	onNoID func(),
```

Comment:
Instead of taking a lambda function here I think it'd be better to just return a result type from `streamRequest`? Even just bool would do for now. Mainly to avoid allocation a function each time to pass to `streamRequest`.

Reply:
:+1: good idea